promql/engine.go

```diff
@@ -695,20 +692,25 @@ func (ng *Engine) populateSeries(ctx context.Context, querier storage.Querier, s
 				hints.End = hints.End - offsetMilliseconds
 			}
 
-			set, wrn, err = querier.Select(false, hints, n.LabelMatchers...)
-			warnings = append(warnings, wrn...)
-			if err != nil {
-				level.Error(ng.logger).Log("msg", "error selecting series set", "err", err)
+			if err := manager.Add(func() (storage.Warnings, error) {
```

Comment:
Let's not change this in any form, we can just do normal select here but for `lazyQuerier.Select`

Now lazyQuerier can do exactly the "add" does, just remember the select params for future and  if lazyQuerier is Sequential, then do sequential one.